src/cli/cmd/mod.rs

```diff
@@ -367,6 +382,99 @@ macro_rules! flakehub_url {
     }};
 }
 
+fn is_root_user() -> bool {
+    nix::unistd::getuid().is_root()
+}
+
+async fn nix_command(args: &[&str], sudo_if_necessary: bool) -> Result<(), FhError> {
+    command_exists("nix")?;
+
+    let use_sudo = sudo_if_necessary && !is_root_user();
+
+    let mut cmd = if use_sudo {
+        tracing::warn!(
+            "Current user is {} rather than root; running Nix command using sudo",
+            whoami::username()
+        );
+        tracing::debug!(
+            "Running: sudo nix --extra-experimental-features 'nix-command-flakes' {}",
+            args.join(" ")
+        );
+
+        let mut cmd = tokio::process::Command::new("sudo");
+        cmd.arg("nix");
+        cmd
+    } else {
+        tracing::debug!(
+            "Running: nix --extra-experimental-features 'nix-command-flakes' {}",
+            args.join(" ")
+        );
+
+        tokio::process::Command::new("nix")
+    };
+
+    let output = cmd
+        .args(["--extra-experimental-features", "nix-command flakes"])
+        .args(args)
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
+        .spawn()
+        .wrap_err("failed to spawn Nix command")?
+        .wait_with_output()
+        .await
+        .wrap_err("failed to wait for Nix command output")?;
+
+    if output.status.success() {
+        Ok(())
+    } else {
+        Err(FhError::FailedNixCommand)
```

Comment:
I think we should include the nix command we ran in the error message; otherwise, it's only visible in debug levels.